libcaf_core/caf/flow/op/interval.cpp

```diff
@@ -87,12 +43,16 @@ class interval_sub : public interval_sub_base {
         last_ = ctx_->steady_time() + initial_delay_;
       else
         last_ = ctx_->steady_time() + period_;
-      pending_ = ctx_->delay_until(last_,
-                                   action{make_counted<interval_action>(this)});
+      schedule_next(last_);
     }
   }
 
-  void fire(interval_action* act) override {
+  void schedule_next(coordinator::steady_time_point timeout) {
+    auto act = make_single_shot_action([this] { fire(); });
+    pending_ = ctx_->delay_until(timeout, std::move(act));
```

Comment:
I believe one of the guidelines was to avoid variables where possible?
```suggestion
    pending_ = ctx_->delay_until(timeout,
                                 make_single_shot_action([this] { fire(); }));
```